### PR TITLE
Folding sheets

### PR DESCRIPTION
**Pull request by LagannPM**
default
Bedsheets can be folded and rolled
Added bedsheet/resting interactions

### CODE REVIEW

**Comment by NanakoAC** [comment]
These are the annoying sheets that get picked up when you click on them right?

Can you make folding the default action instead of alt click, and implement drag and drop to pick them up.
(also maybe possibly have them be placed on a bed when you use them on it, or a mob lying in the bed)

**Comment by LagannPM** [comment]
Had to rename pickup() proc

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -127,15 +127,23 @@
 	return ..(user, distance, "", message)
 
 /obj/item/attack_hand(mob/user as mob)
-	var/atom/old_loc = src.loc
-
-	src.pickup(user)
+	src.before_pickup(user)
```
Comment (NanakoAC) [inline code/game/objects/items.dm]: This doesn't fit with established design patterns, look at pre_equip for an example

It should be called pre_pickup, and it should return a value which dictates whether or not the pickup action completes.

something like
if (!pre_pickup(user))
return

**Comment by LagannPM** [comment]
done

**Pull request merged**